pkg/crc/machine/kubeconfig.go

```diff
@@ -87,6 +79,21 @@ func writeKubeconfig(ip string, clusterConfig *types.ClusterConfig) error {
 	return clientcmd.WriteToFile(*cfg, kubeconfig)
 }
 
+func getGlobalKubeConfig() (string, *api.Config, error) {
+	kubeconfig := getGlobalKubeConfigPath()
+	dir := filepath.Dir(kubeconfig)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return "", nil, err
+	}
+	// Make sure .kube/config exist if not then this will create
+	_, _ = os.OpenFile(kubeconfig, os.O_RDONLY|os.O_CREATE, 0600)
```

Comment:
I'd write this as 
```
    kubeconfig := getGlobalKubeConfigPath()
    if !crcos.FileExists(kubeconfig) {
        dir := filepath.Dir(kubeconfig)
	if err := os.MkdirAll(dir, 0755); err != nil {
		return "", nil, err
	}
	_, _ = os.OpenFile(kubeconfig, os.O_RDONLY|os.O_CREATE, 0600)
    }
    cfg, err := clientcmd.LoadFromFile(kubeconfig)
    if err != nil {
        return "", nil, err
    }
```
This makes it more obvious why mkdirall and OpenFile (I guess this could even go in a helper?) If you use OpenFile(), you need a maching `defer Close()`
k8s.io/client-go handles the same situation slightly differently, might be better to do the same?
```
func getConfigFromFile(filename string) (*clientcmdapi.Config, error) {
	config, err := LoadFromFile(filename)
	if err != nil && !os.IsNotExist(err) {
		return nil, err
	}
	if config == nil {
		config = clientcmdapi.NewConfig()
	}
	return config, nil
}
```